runtime/j9vm/java11vmi.c

```diff
@@ -851,12 +915,26 @@ JVM_DefineModule(JNIEnv * env, jobject module, jboolean isOpen, jstring version,
  * 4) Package is not defined for fromModule's class loader
  * 5) Package is not in module fromModule.
  */
+#if JAVA_SPEC_VERSION >= 15
+void JNICALL
+JVM_AddModuleExports(JNIEnv * env, jobject fromModule, jstring packageObj, jobject toModule)
+#else
 void JNICALL
 JVM_AddModuleExports(JNIEnv * env, jobject fromModule, const char *package, jobject toModule)
+#endif /* JAVA_SPEC_VERSION >= 15 */
 {
 	J9VMThread * const currentThread = (J9VMThread*)env;
 	J9JavaVM const * const vm = currentThread->javaVM;
 	J9InternalVMFunctions const * const vmFuncs = vm->internalVMFunctions;
+#if JAVA_SPEC_VERSION >= 15
+	const char* package = (const char*)(*env)->GetStringUTFChars(env, packageObj, NULL);
+				
+	if (NULL == package) {
+		vmFuncs->setNativeOutOfMemoryError(currentThread, 0, 0);
+		goto done;
+	}
+	convertPackageName((char*)package);
```

Comment:
In OpenJ9, `package` is a private copy, but normally, casting the `const` away is a bad idea.

Reply:
I'm going to ignore this for now. It is possible to rewrite the code to use internal calls instead of GetStringUTFChars().

Reply:
You also can't call setNativeOutOfMemoryError from JNI without entering and exitting the VM. I suggest either writing this in real JNI or entering at the start of the native and writing it as internal VM code.